clippy_lints/src/use_self.rs

```diff
@@ -137,6 +135,14 @@ impl<'tcx> LateLintPass<'tcx> for UseSelf {
     }
 
     fn check_impl_item(&mut self, cx: &LateContext<'_>, impl_item: &hir::ImplItem<'_>) {
+        // Checking items of `impl Self` blocks in which macro expands into.
+        // HACK: Nested macros sometimes clears `macro_backtrace` of types in `check_ty`.
+        // To work around this problem, we check if this impl_item is from macro expansions.
+        // Just remember to remove this state again in `check_impl_item_post`.
```

Comment:
I'd say remove the `HACK`/`macro_backtrace` parts as macros are working as intended here, the issue isn't that a span's context is being cleared but that the same token is being used in two different contexts